mitodl/ocw-studio
# Topics UI and backend

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of  #466 

#### What's this PR do?
 - Adds a new `hierarchical-select` widget
 - Adds topics data to `ocw-course-site-config.yml`
 - Adds management command to generate list of topics to be pasted into the starter config

See RFC for more detail: https://docs.google.com/document/d/1nwHlXF8tpN1aT0awDP5lhJwixKXrHREy0wBrLLqar8U/edit

#### How should this be manually tested?
 - Update the ocw-course starter with the data in `ocw-course-site-config.yml`
 - Go the metadata for a website. You should see a field named "Topics" toward the bottom with no data.
 - Select a topic using the first of the three select fields. If you click "Add Topic" you should see that topic added below. The select fields should also be cleared.
 - If you select a topic, you should see subtopics in the next select field, but only relevant for that topic.
 - If you select a different topic, select a subtopic, and click "Add Topic", you should see something like `topic - subtopic`.
 - If you select a different topic while a subtopic or specialty is selected, it should clear the subtopic or specialty.
 - If you add a duplicate item, it should only show one item and it should not be 
 - If click Save and refresh the browser, it should show the same items which were there before

#### Screenshots
![Screenshot from 2021-08-09 15-37-49](https://user-images.githubusercontent.com/863262/128764136-2dd1f6c3-6a23-44b3-8175-54ab57f79099.png)


![Screenshot from 2021-08-09 15-19-09](https://user-images.githubusercontent.com/863262/128762678-1191bd84-d531-45d8-80eb-8f0d8daab658.png)
